src/exporter.py

```diff
@@ -1,152 +1,85 @@
-# pylint: disable=protected-access,,attribute-defined-outside-init
-import re
+from typing import Any, Callable, Dict
 
 from celery import Celery
-from loguru import logger
-from prometheus_client import CollectorRegistry, Counter, Gauge
 
+from .constants import EventType
+from .event_handlers import (
+    TaskEventHandler,
+    TaskStartedEventHandler,
+    WorkerHeartbeatEventHandler,
+    WorkerStatusEventHandler,
+)
 from .http_server import start_http_server
+from .instrumentation import (
+    celery_worker_up_gauge,
+    queuing_time_gauge,
+    registry,
+    task_failed_event_counter,
+    task_received_event_counter,
+    task_rejected_event_counter,
+    task_retried_event_counter,
+    task_revoked_event_counter,
+    task_sent_event_counter,
+    task_started_event_counter,
+    task_succeeded_event_counter,
+    worker_tasks_active_gauge,
+)
 
 
 class Exporter:
     state = None
 
     def __init__(self):
-        self.registry = CollectorRegistry(auto_describe=True)
-        self.state_counters = {
-            "task-sent": Counter(
-                "celery_task_sent",
-                "Sent when a task message is published.",
-                [
-                    "name",
-                    "hostname",
-                ],
-                registry=self.registry,
+        self.app = None
+
+    def get_handlers(self) -> Dict[str, Callable]:
```

Comment:
I think these should be injected to further decouple these objects but do not expect to have any other handler (celery does not have them) so it should be fine this way.